exoscale/datasource_exoscale_domain_test.go

```diff
@@ -0,0 +1,45 @@
+package exoscale
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccDatasourceDomain(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`
+%s
+data "exoscale_domain" "domain" {
+  name = "${exoscale_domain.exo.name}"
+}`, testAccResourceDomainRecordConfigCreate),
+				Check: resource.ComposeTestCheckFunc(
+					testAccDatasourceDomainAttributes(testAttrs{
+						"name": ValidateString(testDomain),
```

Comment:
Starting from #35 this won't work anymore: we don't have a global `testDomain` variable with a static domain name, we now generate a random domain name at every acceptance test run.

You should do something in this spirit:

```go
var testAccDatasourceDomainName = testPrefix + "-" + testRandomString() + ".net"

func TestAccDatasourceDomain(t *testing.T) {
	resource.Test(t, resource.TestCase{
		PreCheck:  func() { testAccPreCheck(t) },
		Providers: testAccProviders,
		Steps: []resource.TestStep{
			{
				Config: fmt.Sprintf(`
resource "exoscale_domain" "test" {
  name = "%s"
}

data "exoscale_domain" "test" {
  name = exoscale_domain.test.name
}`, testAccDatasourceDomainName),
				Check: resource.ComposeTestCheckFunc(
					testAccDatasourceDomainAttributes(testAttrs{
						"name": ValidateString(testAccDatasourceDomainName),
					}),
				),
			},
		},
	})
}
```

Reply:
thanks !